### ISSUE TITLE
eng.lstm is not a recognition model?

### ISSUE THREAD

**Issue by ckcollab** [artdevgame/tesseract-trainer#4]
Trying this from WSL + WSL Docker and I am getting...

```bash
❯ docker-compose up
Starting train-ocr ... done
Attaching to train-ocr
train-ocr | Cloning into 'langdata_lstm'...
Checking out files: 100% (1237/1237), done.237)
train-ocr | Fetching updates from the langdata_lstm repo (be patient, you'll see a message when done)
train-ocr | From https://github.com/tesseract-ocr/langdata_lstm
train-ocr |  * branch            HEAD       -> FETCH_HEAD
train-ocr | Already up to date.
train-ocr | Fetch complete
train-ocr | start.sh: 17: start.sh: tesstrain.sh: not found
train-ocr | Version string:4.00.00alpha:eng:synth20170629:[1,36,0,1Ct3,3,16Mp3,3Lfys64Lfx96Lrx96Lfx512O1c1]
train-ocr | 17:lstm:size=11689099, offset=192
train-ocr | 18:lstm-punc-dawg:size=4322, offset=11689291
train-ocr | 19:lstm-word-dawg:size=3694794, offset=11693613
train-ocr | 20:lstm-number-dawg:size=4738, offset=15388407
train-ocr | 21:lstm-unicharset:size=6360, offset=15393145
train-ocr | 22:lstm-recoder:size=1012, offset=15399505
train-ocr | 23:version:size=80, offset=15400517
train-ocr | Extracting tessdata components from /usr/local/share/tessdata/eng.traineddata
train-ocr | Wrote /app/src/train/eng.lstm
train-ocr | /app/src/train/eng.lstm is not a recognition model, trying training checkpoint...
train-ocr | Failed to load list of eval filenames from /app/src/train/eng.training_files.txt
train-ocr | Failed to load eval data from: /app/src/train/eng.training_files.txt
train-ocr | Failed to load list of training filenames from /app/src/train/eng.training_files.txt
train-ocr | Failed to read continue from: /app/src/output/_checkpoint
train-ocr | /app/src/output/_checkpoint is not a recognition model, trying training checkpoint...
train-ocr | Failed to load model from: /app/src/output/_checkpoint
```

looks like a problem with the generated `eng.lstm`? Maybe windows file permissions or something?

```bash
❯ ls -aln src/train
total 11424
drwxr-xr-x 2    0    0     4096 Aug 22 18:19 .
drwxr-xr-x 6 1000 1000     4096 Aug 22 18:19 ..
-rw-r--r-- 1    0    0 11689099 Aug 22 18:19 eng.lstm
```

_EDIT:_

I modified the `Dockerfile` and now the file belongs to my user `1000:1000`, but it still fails the same way, so I suppose not a permissions issue?

**Comment by ckcollab** [comment]
Actually, @artdevgame I'm getting this same error on my Mac as well.. wonder if it's something else that is the problemo?

**Comment by ministerofpanic** [comment]
Hey @ckcollab 

Sorry for the late reply, I don't seem to be getting notifications.
Did you manage to resolve this?

In your logs I noticed:
```
train-ocr | start.sh: 17: start.sh: tesstrain.sh: not found
```

Which is coming from this line:
https://github.com/artdevgame/tesseract-trainer/blob/master/start.sh#L17

The file should exist in `/app/src`, which is the working directory inside the container:
https://github.com/artdevgame/tesseract-trainer/blob/master/start.sh#L1

I think [that file](https://github.com/svn2github/tesseract-ocr/blob/master/training/tesstrain.sh) is created when cloning the tesseract-ocr project:
https://github.com/artdevgame/tesseract-trainer/blob/master/start.sh#L5

**Comment by Malkiz223** [comment]
I got the same error as ckcollab.
@artdevgame

At first I was getting  (space before .git/)
```
train-ocr  | start.sh: 1: cd: can't cd to /app/src
: not found| start.sh: 2: start.sh:
train-ocr  | Cloning into 'langdata_lstm'...
train-ocr  | fatal: unable to access 'https://github.com/tesseract-ocr/langdata_lstm .git/': The requested URL returned error: 400
train-ocr  | start.sh: 9: start.sh: Syntax error: word unexpected
 repo (be patient, you'll see a message when done)tm
train-ocr exited with code 2 
```

To fix this I had to change in the start.sh file
LSTM_REMOTE="https://github.com/tesseract-ocr/$LSTM_REPO.git"
on
LSTM_REMOTE="https://github.com/tesseract-ocr/$LSTM_REPO"


Then I started getting the same error:
```
[+] Running 2/2
 - Network tesseract-trainer_default  Created                                                                                                                                                                                      0.7s
 - Container train-ocr                Created                                                                                                                                                                                      0.3s
Attaching to train-ocr
train-ocr  | Fetching updates from the langdata_lstm repo (be patient, you'll see a message when done)
train-ocr  | From https://github.com/tesseract-ocr/langdata_lstm
train-ocr  |  * branch            HEAD       -> FETCH_HEAD
train-ocr  | Already up to date.
train-ocr  | Fetch complete
train-ocr  | start.sh: 17: start.sh: tesstrain.sh: not found
train-ocr  | Version string:4.00.00alpha:eng:synth20170629:[1,36,0,1Ct3,3,16Mp3,3Lfys64Lfx96Lrx96Lfx512O1c1]
train-ocr  | 17:lstm:size=11689099, offset=192
train-ocr  | 18:lstm-punc-dawg:size=4322, offset=11689291
train-ocr  | 19:lstm-word-dawg:size=3694794, offset=11693613
train-ocr  | 20:lstm-number-dawg:size=4738, offset=15388407
train-ocr  | 21:lstm-unicharset:size=6360, offset=15393145
train-ocr  | 22:lstm-recoder:size=1012, offset=15399505
train-ocr  | 23:version:size=80, offset=15400517
train-ocr  | Extracting tessdata components from /usr/local/share/tessdata/eng.traineddata
train-ocr  | Wrote /app/src/train/eng.lstm
train-ocr  | /app/src/train/eng.lstm is not a recognition model, trying training checkpoint...
train-ocr  | Failed to load list of eval filenames from /app/src/train/eng.training_files.txt
train-ocr  | Failed to load eval data from: /app/src/train/eng.training_files.txt
train-ocr  | Failed to load list of training filenames from /app/src/train/eng.training_files.txt
train-ocr  | Failed to read continue from: /app/src/output/_checkpoint
train-ocr  | /app/src/output/_checkpoint is not a recognition model, trying training checkpoint...
train-ocr  | Failed to load model from: /app/src/output/_checkpoint
train-ocr exited with code 1
```
I also use WSL2 and Docker WSL on Windows, but not sure if this is the problem.

**Comment by ministerofpanic** [comment]
Hey @ckcollab, @Malkiz223,

I finally got some time to take a look at this. TL;DR: tesseract is now on v5, and these scripts are setup for v4.

In v5, the tesstrain.sh file was removed:
- https://github.com/tesseract-ocr/tesseract/commit/93348a83a324a4
- https://github.com/tesseract-ocr/tesseract/issues/3455

I haven't figured out the correct upgrade path yet, so for now I'm pinning `tesseract-ocr/tesseract` to `4.1.3`, this will allow the project to build as before.

If you've still got this installed and need to update, make sure you run `docker-compose build`
